pkg/mimirpb/compat_rw2.go

```diff
@@ -56,12 +57,21 @@ func (ps *rw2PagedSymbols) releasePages() {
 }
 
 func (ps *rw2PagedSymbols) get(ref uint32) (string, error) {
+	if ref < ps.offset {
+		if len(ps.commonSymbols) == 0 {
+			return "", fmt.Errorf("symbol %d is under the offset %d, but no common symbols table was registered", ref, ps.offset)
+		}
+		if ref > uint32(len(ps.commonSymbols)) {
```

Comment:
Shouldn't the check be >=? For example, if ref=1 and the len=1, then it should fail.

```suggestion
		if ref >= uint32(len(ps.commonSymbols)) {
```